File: hanoi/hanoi.js
function hanoi(n, a, b) {
  var temp;
  if(n!=1) {
    temp=6-a-b;
    hanoi(n-1, a, temp);
    hanoi(n-1, temp, b);
  }
}

var st = new Date().getTime();
hanoi(20, 1, 2);
var t = new Date().getTime() - st;
console.log(t + " ms");

File: hanoi/hanoi.r
hanoi <- function(n, a, b) {
  if(n!=1) {
    temp=6-a-b
    hanoi(n-1,a,temp)
    hanoi(n-1,temp,b)
  }
}

args=(commandArgs(TRUE))
n = as.numeric(args[1])

hanoi(n, 1, 2)